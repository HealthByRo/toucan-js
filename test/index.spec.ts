import Toucan from "../src/index";
import {
  makeEvent,
  getFetchMockPayload,
  mockServiceWorkerEnv,
  mockFetch,
  mockDateNow,
  resetDateNow,
} from "./helpers";

const VALID_DSN = "https://123:456@testorg.ingest.sentry.io/123";

describe("Toucan", () => {
  beforeEach(() => {
    mockServiceWorkerEnv();
    mockFetch();
    mockDateNow();
    jest.resetModules();
  });

  afterEach(() => {
    resetDateNow();
  });

  test("disabled mode", async () => {
    const event = makeEvent();

    // Valid option, set disabled = true
    const toucan1 = new Toucan({
      dsn: "",
      event,
    });
    // Valid option, set disabled = true
    const toucan2 = new Toucan({
      event,
    } as any);
    // Invalid option, log 'SentryError: Invalid Dsn', and set disabled = true
    const toucan3 = new Toucan({
      dsn: "hello",
      event,
    });

    expect(toucan1.captureMessage("test")).toBeUndefined();
    expect(toucan2.captureMessage("test")).toBeUndefined();
    expect(toucan3.captureMessage("test")).toBeUndefined();
  });

  test("captureException", async () => {
    const event = makeEvent();
    const toucan = new Toucan({
      dsn: VALID_DSN,
      event,
    });

    expect(toucan.captureMessage("test")).toHaveLength(32);
    expect(global.fetchMock).toHaveBeenCalledTimes(1);
    expect(getFetchMockPayload(global.fetchMock)).toMatchSnapshot();
  });
});
