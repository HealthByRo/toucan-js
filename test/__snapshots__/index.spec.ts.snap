// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toucan captureException 1`] = `
Object {
  "breadcrumbs": Array [],
  "event_id": "651b177fe1cb4ac89e15c1ecd2cb1d0a",
  "level": "info",
  "logger": "EdgeWorker",
  "message": "test",
  "platform": "node",
  "request": Object {
    "method": "GET",
    "url": "https://example.com/",
  },
  "sdk": Object {
    "name": "__name__",
    "version": "__version__",
  },
  "timestamp": 1586752837.868,
}
`;
